Gradle sample not working (timeout error)

<!--- Provide a general summary of the issue in the Title above -->

## Expected Behavior
<!--- Tell us what should happen -->

I expect to be able to run the [gradle sample](https://github.com/paketo-buildpacks/samples/tree/main/java/gradle) out-of-the-box and have it complete successfully.

## Current Behavior
<!--- Tell us what happens instead of the expected behavior -->

When the gradle buildpack runs during the sample build, I receive the error below.

```
[2023-10-24T20:43:07.992Z] [builder]   Compiled Application: Contributing to layer
[2023-10-24T20:43:07.992Z] [builder]     Executing gradlew --no-daemon -Dorg.gradle.welcome=never assemble
[2023-10-24T20:43:08.246Z] [builder]       Downloading https://services.gradle.org/distributions/gradle-7.5-bin.zip
[2023-10-24T20:43:08.500Z] [builder]       
[2023-10-24T20:43:08.500Z] [builder]       Exception in thread "main" java.io.IOException: Downloading from https://services.gradle.org/distributions/gradle-7.5-bin.zip failed: timeout (1ms)
[2023-10-24T20:43:08.500Z] [builder]       	at org.gradle.wrapper.Download.downloadInternal(Download.java:131)
[2023-10-24T20:43:08.500Z] [builder]       	at org.gradle.wrapper.Download.download(Download.java:89)
[2023-10-24T20:43:08.500Z] [builder]       	at org.gradle.wrapper.Install$1.call(Install.java:83)
[2023-10-24T20:43:08.500Z] [builder]       	at org.gradle.wrapper.Install$1.call(Install.java:63)
[2023-10-24T20:43:08.500Z] [builder]       	at org.gradle.wrapper.ExclusiveFileAccessManager.access(ExclusiveFileAccessManager.java:69)
[2023-10-24T20:43:08.500Z] [builder]       	at org.gradle.wrapper.Install.createDist(Install.java:63)
[2023-10-24T20:43:08.500Z] [builder]       	at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:109)
[2023-10-24T20:43:08.500Z] [builder]       	at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:66)
[2023-10-24T20:43:08.500Z] [builder]       Caused by: java.net.SocketTimeoutException: Connect timed out
[2023-10-24T20:43:08.500Z] [builder]       	at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:551)
[2023-10-24T20:43:08.500Z] [builder]       	at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:602)
[2023-10-24T20:43:08.500Z] [builder]       	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
[2023-10-24T20:43:08.500Z] [builder]       	at java.base/java.net.Socket.connect(Socket.java:633)
[2023-10-24T20:43:08.500Z] [builder]       	at java.base/sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:304)
[2023-10-24T20:43:08.500Z] [builder]       	at java.base/sun.net.NetworkClient.doConnect(NetworkClient.java:178)
[2023-10-24T20:43:08.500Z] [builder]       	at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:533)
[2023-10-24T20:43:08.500Z] [builder]       	at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:638)
[2023-10-24T20:43:08.500Z] [builder]       	at java.base/sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:266)
[2023-10-24T20:43:08.500Z] [builder]       	at java.base/sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:380)
[2023-10-24T20:43:08.500Z] [builder]       	at 
```

## Possible Solution
<!--- Not obligatory, but suggest an acceptable fix for the bug -->

Speaking with @anthonydahanne in a [Slack thread](https://paketobuildpacks.slack.com/archives/C0124SD3GTG/p1698258102926649), he noted that there was a [recent change  in the timeout](https://github.com/paketo-buildpacks/samples/pull/585/files#diff-d873c1943a34cb10faa94d7520d4a99b561554eecacf6106b4c65482d6a4926fL4) that the gradle wrapper uses (this explains why it was working for so long for us, but recently broke).

He suggested I change the timeout from `1` (ms) to `10000`. I tested this manually and my builds worked.

## Steps to Reproduce
<!--- Provide a link to an reproduction/test case, or an unambiguous set of steps to reproduce this bug. Include notes about the software versions you're using, the environment in which you're running, code snippets to reproduce, log output, screenshots, etc.. -->

I was able replicate the issue on multiple machines using the following commands. I did this on multiple machines in my company's AWS account, on a company Mac off the company VPN (using `pack` installed via `brew`), and on my personal Linux box at home.

```
# install pack
(curl -sSL "https://github.com/buildpacks/pack/releases/download/v0.30.0/pack-v0.30.0-linux.tgz" | sudo tar -C /usr/local/bin/ --no-same-owner -xzv pack)

# clone sample repo
git clone https://github.com/paketo-buildpacks/samples
cd samples/java/gradle

# build the sample (last command throws the error)
/usr/local/bin/pack config default-builder paketobuildpacks/builder-jammy-base
/usr/local/bin/pack build applications/gradle
```

## Motivations
<!--- How has this issue affected you? What are you trying to accomplish? What is the impact? Providing context helps us come up with a solution that is most useful in the real world. -->

We're building our buildpacks in a CI/CD Jenkins pipeline and pushing them to our internal AWS ECR. We're using the samples to test our builds and this sample recently started to fail. I would expect the sample to run out-of-the-box. I'm not a Java developer, so I have limited knowledge of where to start troubleshooting.

--- a/.github/workflows/java-gradle.yml
+++ b/.github/workflows/java-gradle.yml
@@ -29,5 +29,5 @@ jobs:
           key: ${{ runner.os }}-gradle-${{ hashFiles('java/gradle/*.gradle') }}
           restore-keys: ${{ runner.os }}-gradle
       # simulate the bindings
-      - run: cp bindings/gradle-wrapper/gradle-wrapper.properties gradle/wrapper/gradle-wrapper.properties && ./gradlew build
+      - run: ./gradlew build
         working-directory: java/gradle

--- a/java/gradle/gradle/wrapper/gradle-wrapper.properties
+++ b/java/gradle/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 distributionUrl=https\://services.gradle.org/distributions/gradle-7.5-bin.zip
-networkTimeout=1
+networkTimeout=10000
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists

